src/hooks/use-content-folders.ts

```diff
@@ -0,0 +1,60 @@
+import * as Sentry from '@sentry/electron/renderer';
+import { useState, useEffect } from 'react';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+export type WpContentFolder = 'plugins' | 'themes';
+export type WpContentFolderEntry = { name: string; type: 'file' | 'folder' };
+export type WpContentFolderResult = {
+	items: WpContentFolderEntry[];
+	isLoading: boolean;
+	error: Error | null;
+};
+
+export function useContentFolders(
+	siteId: string,
+	types: WpContentFolder[]
+): Record< WpContentFolder, WpContentFolderResult > {
+	const [ results, setResults ] = useState< Record< WpContentFolder, WpContentFolderResult > >(
+		() => {
+			const initialResults: Record< WpContentFolder, WpContentFolderResult > = {
+				plugins: { items: [], isLoading: true, error: null },
+				themes: { items: [], isLoading: true, error: null },
+			};
+
+			return initialResults;
+		}
+	);
```

Comment:
Hm, coudl you please explain this way of defining default lvalue? Instead of classic way as:
```suggestion
const [ results, setResults ] = useState< Record< WpContentFolder, WpContentFolderResult > >( {
		plugins: { items: [], isLoading: true, error: null },
		themes: { items: [], isLoading: true, error: null },
	} );
```